x/programs/rust/wasmlanche-sdk/src/state.rs

```diff
@@ -38,8 +42,15 @@ pub enum Error {
     Delete,
 }
 
+#[derive(Debug, PartialEq)]
+pub enum CachedData {
+    Data(Vec<u8>),
+    Deleted,
+    ToDelete,
+}
```

Comment:
Here, `Deleted` means "we know that there is no value here", it could be a bit unclear since it may mean that some value was here before, while it just means that we checked in the actual storage and there is no value.

Reply:
Also, having a value at a key `K` in the cache means "we know the state of the value here". If it's not here that means that it's unknown.